test/prob/test_fixture_ccdf_log.hpp

```diff
@@ -582,124 +563,170 @@ class AgradCcdfLogTestFixture : public ::testing::Test {
-            pos_single, multiple_gradients2, pos_multiple, N_REPEAT);
-
-      pos_single = 0;
-      pos_multiple = 0;
-      if (!is_constant_all<T0>::value && !is_empty<T0>::value
-          && std::is_same<Scalar0, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T0>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T1>::value && !is_empty<T1>::value
-          && std::is_same<Scalar1, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T1>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T2>::value && !is_empty<T2>::value
-          && std::is_same<Scalar2, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T2>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T3>::value && !is_empty<T3>::value
-          && std::is_same<Scalar3, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T3>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T4>::value && !is_empty<T4>::value
-          && std::is_same<Scalar4, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T4>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
-      if (!is_constant_all<T5>::value && !is_empty<T5>::value
-          && std::is_same<Scalar5, fvar<fvar<var>>>::value)
-        test_multiple_gradient_values(
-            is_vector<T5>::value, single_ccdf_log, single_gradients3,
-            pos_single, multiple_gradients3, pos_multiple, N_REPEAT);
+      EXPECT_EQ(single_gradients1.size(), multiple_gradients1.size())
+          << "scalar and vectorized results should produce the same number of "
+             "first order gradients";
+      EXPECT_EQ(single_gradients2.size(), multiple_gradients2.size())
+          << "scalar and vectorized results should produce the same number of "
+             "second order gradients";
+      EXPECT_EQ(single_gradients3.size(), multiple_gradients3.size())
+          << "scalar and vectorized results should produce the same number of "
+             "third order gradients";
+
+      for (size_t i = 0; i < single_gradients1.size(); ++i)
+        EXPECT_NEAR(single_gradients1[i], multiple_gradients1[i], 1e-7)
+            << "scalar and vectorized results should have the same first order "
+               "gradients";
+      for (size_t i = 0; i < single_gradients2.size(); ++i)
+        EXPECT_NEAR(single_gradients2[i], multiple_gradients2[i], 1e-7)
+            << "scalar and vectorized results should have the same first order "
```

Comment:
```suggestion
            << "scalar and vectorized results should have the same second order "
```

Reply:
While this simplifies the code, the error message loses the information which variable has wrong gradient.

Reply:
This error information wasn't printed anywhere anyway.

Reply:
Oh, right I added it in https://github.com/stan-dev/math/pull/2082, but it is not merged yet. Could you print it? That is very useful info for debugging. (since it is not in yet lets make this optional)

Reply:
Just a minor note: you might want to use `expect_near_rel` for inexact comparisons - we thought a bit about good logic for this type of comparisons and also it works out of the box for vector/matrix arguments (see e.g. https://github.com/stan-dev/math/issues/1656)

Reply:
@martinmodrak switched to expect_near_rel and life did get easier, thanks for rec.